assets/js/components/wp-dashboard/WPDashboardPopularPages.js

```diff
@@ -110,10 +109,15 @@ export default function WPDashboardPopularPages( props ) {
 		return <WidgetReportError moduleSlug="analytics" error={ error } />;
 	}
 
-	if ( isZeroReport( report ) && ! zeroDataStatesEnabled ) {
+	if ( isGatheringData && ! zeroDataStatesEnabled ) {
 		return <WidgetReportZero moduleSlug="analytics" />;
 	}
 
+	// Skip rendering the table if there are no rows.
+	if ( ! report[ 0 ].data?.rows?.length ) {
+		return null;
+	}
```

Comment:
In my case, the `report` object was in tact but `data.rows` was `undefined` which raised the error below when trying to call `forEach` on `undefined`.

This seemed like a better option than adding an empty array as a fallback as the table would have headers but no rows.